### ISSUE TITLE
Feature request - Adding Proper TF 2.0 Checkpoints (not onnx-tensorflow) + Batching + TF JS

### ISSUE THREAD

**Issue by aleks73337** [snakers4/silero-models#5]
Hello, gyus! Your models are brilliant and I want to use it in my project via tensorflow serving. But it can't works without batching. Can you pleese save models with batching? Thank you!

**Comment by snakers4** [comment]
Tbh, I am not that familiar with modern Tensorflow.
I exported from ONNX to Tensorflow via provided export utils and when I tried loading it into plain Tensorflow, it failed with cryptic errors.
Were you able to load the current models? If so, how?
Adding batching is not very difficult, but it would make sense in case you are actually able to load models.

**Comment by aleks73337** [comment]
I converted model from onnx to tensorflow and it worked. Here's the code:


`
from onnx_tf.backend import prepare
import json
import onnx
import torch
import tempfile
import onnxruntime
from omegaconf import OmegaConf
from utils import *

models = OmegaConf.load('models.yaml')

with tempfile.NamedTemporaryFile('wb', suffix='.json') as f:
    torch.hub.download_url_to_file(models.stt_models.en.latest.labels,
                                   f.name,
                                   progress=True)
    with open(f.name) as f:
        labels = json.load(f)
        decoder = Decoder(labels)

with tempfile.NamedTemporaryFile('wb', suffix='.model') as f:
    torch.hub.download_url_to_file(models.stt_models.en.latest.onnx,
                                   f.name,
                                   progress=True)
    onnx_model = onnx.load(f.name)
    onnx.checker.check_model(onnx_model)
    ort_session = onnxruntime.InferenceSession(f.name)
tf_rep = prepare(onnx_model)
tf_rep.export_graph('saved_model')`
After that model will be in saved_model dir

**Comment by aleks73337** [comment]
When I tried to load tensorflow saved model, I got the crytical error too

**Comment by snakers4** [comment]
Well, I essentially did the same bit:

```
tf_rep = prepare(onnx_model)
tf_rep.export_graph('saved_model')`
```

But I have been using PyTorch exclusively for a couple of years now, and the only hypothesis that I have - is that TF changed their serialization formats some time ago. I googled a bit and I was not able to find anything,

Please note that I am not a big expert in TF. 
Maybe someone here has more expertise in TF + ONNX.

**Comment by snakers4** [comment]
Also as the examples show, I actually tested the ONNX model via onnx runtime and it works.
So I am sure that the problem is with export.

**Comment by aleks73337** [comment]
![image](https://user-images.githubusercontent.com/32843048/93858006-2b643b00-fcc4-11ea-9d2f-8dbb0b88f2fe.png)
Theese 2 folders are important too. TF can't load model without them. But you provided only .pb file.

**Comment by aleks73337** [comment]
For loading model from saved dir, use 
`imported =  tf.keras.models.load_model("saved_model")`

**Comment by snakers4** [comment]
let me try saving it as a folder as well

**Comment by snakers4** [comment]
I checked what is actually being exported

- https://github.com/onnx/onnx-tensorflow#onnx-model-inference-with-tensorflow-backend
- https://github.com/onnx/onnx-tensorflow/blob/master/doc/API.md#onnx-tensorflow-api

looks like they provide these conversion utils only for usage with onnx-tensorflow, i.e. this is not a proper TF model
all of this very misleading

**Comment by snakers4** [comment]
I am not really sure what can be done with this / can invest time into this now
Any help / feedback / advice / PR from the community is appreciated

**Comment by aleks73337** [comment]
Can you add batching to ONNX model, sond it to me and I'll convert it to TF and send to you?

**Comment by snakers4** [comment]
Yes, of course we can do that
I have found this [ticket](https://stackoverflow.com/questions/59657166/convert-frozen-model-pb-to-savedmodel)

Are you doing something similar? 
Maybe I am doing something wrong, because `tf_rep.export_graph('saved_model')` does not give me `assets` or `variables`

**Comment by aleks73337** [comment]
Which version of TF and onnx_tf do you use?

**Comment by aleks73337** [comment]
> Are you doing something similar?
No, I just use export_graph and get the right type :)

**Comment by snakers4** [comment]
> Which version of TF and onnx_tf do you use?

tf `2.3.0`
onnx_tf `1.6.0`

as for batched methods - I tried making batched models, but for now I have weird padding issues when checking with `onnxruntime`, so far thinking how to solve them

**Comment by aleks73337** [comment]
I use TF GPU 2.2.0 and onnx_tf 1.6.0. I got a lot of errors when tried to combine them. 
Ok. I'll wait your model with batching :) 
I can send you TF saved model without batching and instruction how to use it if you want :)

**Comment by snakers4** [comment]
> I can send you TF saved model without batching and instruction how to use it if you want :)

that would be helpful for the time being as a workaround

could you please send tar archives for all of the models?
please note that we are not planning major TF support

**Comment by snakers4** [comment]
> I use TF GPU 2.2.0 and onnx_tf 1.6.0. I got a lot of errors when tried to combine them.

is there anything special about this combination?

**Comment by aleks73337** [comment]
> > I use TF GPU 2.2.0 and onnx_tf 1.6.0. I got a lot of errors when tried to combine them.
> 
> is there anything special about this combination?

No, but onnx_tf writes that they support 2.2.0 in last release.

> could you please send tar archives for all of the models?

You mean ONNX and TF?

**Comment by snakers4** [comment]
> No, but onnx_tf writes that they support 2.2.0 in last release.

oh, as usual, I love TF xD

>> could you please send tar archives for all of the models?
> You mean ONNX and TF?

no, I have ONNX =)

I mean that the result of this should be a folder
https://user-images.githubusercontent.com/32843048/93858006-2b643b00-fcc4-11ea-9d2f-8dbb0b88f2fe.png

to add these to the repo (at least for now) one needs to make a tar for each model

> I got a lot of errors when tried to combine them

there is not straightforward way to install them both?

**Comment by aleks73337** [comment]
> there is not straightforward way to install them both?

Try just using pip. Maybe it'll work for you.

Archieve is too high for GitHub. Uploaded it to google drive:

https://drive.google.com/file/d/1uYsAHWEihATJRBofxdOMSE7eIj5DcqmI/view?usp=sharing

**Comment by aleks73337** [comment]
How to use:
`imported =  tf.keras.models.load_model("saved_model")
test_audio = read_audio(random.sample(test_files, k = 1)[0]).numpy()
res = imported.signatures["serving_default"](tf.constant(test_audio))['output_0']
decoder(torch.Tensor(res.numpy()))`

**Comment by snakers4** [comment]
> Archieve is too high for GitHub. Uploaded it to google drive

Looks like this is an English model
Can you please do the same with all of the other models?
This will be very helpful
I will be able also to convert to tf-js as well

**Comment by snakers4** [comment]
I added a SavedModel tf model
added a tf-js model
added an int8 tf-js model (experimental)

**Comment by snakers4** [comment]
Interesting I installed tensorflow `2.2.0`, I have  onnx_tf  `1.6.0`.
And it still exports to a single protobuf file
Is there an additional step maybe?

**Comment by snakers4** [comment]
https://github.com/onnx/onnx-tensorflow/pull/603 
was merged 12 days ago
looks like this was the culprit

**Comment by snakers4** [comment]
as for batched inference - let's put a pin in that and return later
I will convert other two models myself

**Comment by aleks73337** [comment]
> Interesting I installed tensorflow `2.2.0`, I have onnx_tf `1.6.0`.
> And it still exports to a single protobuf file
> Is there an additional step maybe?

No, just 1 function call. I can't understand why it is so
